guides/releasing/Release Workflow.md

```diff
@@ -0,0 +1,328 @@
+  - Make sure Expo client builds in Xcode as expected.
+- **Android**:
+  - Run `et remove-sdk --platform android` to remove Android's `expoview-abiXX_0_0` library and versioned code.
+  - This script will print all references to the SDK version it just removed - check all of them and remove them if possible (remove legacy code).
+  - Make sure Expo client builds in Android Studio as expected.
+- Repeat **iOS** and **Android** specific steps if you want to delete more SDK versions.
+- Commit changes and create a pull request to `master` branch.
+
+## 1.2. Update React Native
+
+**Why:** Each SDK version has its own version of React Native. If we're planning to update React Native version for the upcoming SDK then we need to update our fork of React Native and update `react-native-lab/react-native` submodule in this repo which is the source of truth for `react-native` version used throughout the repository.
+    
+**How:**
+
+- Go to `react-native-lab/react-native` submodule.
+- Coordinate with James (@ide) to create a new branch for new SDK in our `react-native` fork (`sdk-XX` typically, where `XX` is the major number of the SDK version).
+- Run `et update-react-native`. This expotools command copies `ReactAndroid` and `ReactCommon` folders from the submodule to the respective paths: `android/ReactAndroid` and `android/ReactCommon` and then executes `ReactAndroidCodeTransformer` that applies some Expo-specific code transformations.
+- Add your git changes from both `react-native-lab` and `android` folders and create a pull request to `master` branch.
+
+## 1.3. Versioning code for the new SDK
+
+| Requirements                                         |
+| ---------------------------------------------------- |
+| [1.2. Update React Native](#1-2-update-react-native) |
+
+**Why:** As we need to support multiple SDK versions, the code needs to be prefixed for each version so we don't get conflicts and duplicated classes. Such prefixed version of our APIs is called ABI.
+
+**How:**
+
+- **iOS**:
+  - Run `et add-sdk --platform ios` to copy unversioned code into the new ABI and prefix (or suffix) its files and corresponding code references with `ABIXX_0_0`. If this script errors partway, you can always delete the new directory it created under `ios/versioned-react-native` and revert any other changes it made to `EXSDKVersions.plist` and `sdkVersions.json` (or just run `et remove-sdk --platform --sdkVersion XX.0.0`). Then it's safe to run this script again.
+  - Let the `add-sdk` script to regenerate Podfile and reinstall pods, then try to build the project in Xcode. This script does most of the work, but usually breaks in various ways, partly because some assumptions change every SDK cycle. If you found anything broken, please keep versioning script up to date.
+  - Submit a pull request with versioned code to `master` branch.
+- **Android**:
+  - *todo*
```

Comment:
- Run `et add-sdk --platform android` to create the new versioned AAR and expoview code. This script will attempt to rename some native libraries and will ask you to manually verify that it has renamed them all properly. If you notice some that are missing, add them to the list in `tools/expotools/src/versioning/android/libraries.ts` and rerun the script. Commit the changes.
- Run `et android-build-packages` to build the packaged AARs for each unimodule for the new SDK version. If any of the builds fail, follow the directions from the script to fix the build and then rerun the build command for only the failed packages. Commit the changes.